src/polycap-transmission-efficiencies.c

```diff
@@ -768,6 +768,80 @@ bool polycap_transmission_efficiencies_write_hdf5(polycap_transmission_efficienc
 	return true;
 }
 //===========================================
+void polycap_images_free(struct _polycap_images *images)
```

Comment:
@tschoonj Kan je dit even nakijken? Ik versta de LGTM error niet, en ik weet dat valgrind hier een fout ziet (memory lost), maar kzie niet direct hoe het anders moet...

Reply:
Ter verduidelijking: de bedoeling is dat we in Python ook deze images struct kunnen extraheren, zodat we daar ook de coordinaten etc van alle events krijgen.
Natuurlijk werk ik in python liefst met 2D arrays ipv de 1D pointer arrays in C, dus ik begon me af te vragen of ik misschien in Python dan beter een andere class maak, die sterk zal lijken op images natuurlijk. Of is er geen probleem dat de Python images struct wat zou afwijken van de C images struct?

En dan is er ook die TODO nog van bij `Photon.extleak` over cached en return iterable... is dit gewoon gefikst door @lru_cache boven @property te plaatsen? Lijkt me iets te simpel, en mocht het simpel zijn had het vermoedelijk al geimplementeerd geweest :P

Reply:
Ik heb de LGTM check herstart, ik denk dat het een tijdelijk probleem bij hen was.

Uw polycap_images bevat nogal veel andere data. Zijn die allemaal 2D arrays van doubles gerangschikt via row-major ordering? Zou niet moeilijk mogen zijn om die in 2D numpy arrays te krijgen dan.

Die TODO kan waarschijnlijk opgelost worden door gewoon uw individuele Python leaks te `yield`en ipv de hele array van Python leak objects te returnen. Dit moet dan gecombineerd worden met de C array van de leaks te cachen als member variable. Dit zou geheugen gebruik moeten schappelijk houden (tenzij de iterator dan expliciet naar een list wordt gecast, maar dat is dan de user die lomp is 😄 )

Reply:
> Die TODO kan waarschijnlijk opgelost worden door gewoon uw individuele Python leaks te `yield`en ipv de hele array van Python leak objects te returnen. 

Right... dus gewoon `return` vervangen door `yield` (`yield` dan wel binnen die for loop in `Photon.extleak`) en Python magie + onderstaande doet de rest?

> Dit moet dan gecombineerd worden met de C array van de leaks te cachen als member variable. Dit zou geheugen gebruik moeten schappelijk houden 

Snap ik niet... Quick google doet me denken dat ik die rturn values van de C array moet opslaan in een tweede variabele, maar dat lijkt me dan alsof ik net meer memory ga gebruiken (dubbel zoveel) :S 

> (tenzij de iterator dan expliciet naar een list wordt gecast, maar dat is dan de user die lomp is 😄 )

That would be me I guess :D

Reply:
Yep zoiets, zal straks eens zien of ik dat snel kan implementeren. Btw wat gebeurt er als de leaks niet berekend werden (de default denk ik)? Wat doen `polycap_photon_get_extleak_data` en zijn python counterpart dan?

Reply:
normaal return None (als n-leaks < 0)
En als photon zelf NULL is, returned extleak ook None uiteraard.